packages/material-ui-lab/src/Slider/Slider.js

```diff
@@ -319,15 +319,21 @@ class Slider extends React.Component {
   };
 
   emitChange(event, rawValue, callback) {
-    const { step, value: previousValue, onChange, disabled } = this.props;
+    const { step, value: previousValue, onChange, disabled, min, max } = this.props;
     let value = rawValue;
 
     if (disabled) {
       return;
     }
 
     if (step) {
-      value = roundToStep(rawValue, step);
+      if (rawValue > min && rawValue < max) {
+        if (rawValue === max - step) {
+          value = roundToStep(rawValue + step / 2, step);
```

Comment:
This does not consider edge values at the minimum. It would be nice to add some documentation to this. To be honest it took some time for me to understand what is happening here.